sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -58,18 +64,53 @@ case class CreateTableLikeCommand(
       throw new AnalysisException(
         s"Source table in CREATE TABLE LIKE does not exist: '$sourceTable'")
     }
-    if (catalog.isTemporaryTable(sourceTable)) {
-      throw new AnalysisException(
-        s"Source table in CREATE TABLE LIKE cannot be temporary: '$sourceTable'")
-    }
 
-    val tableToCreate = catalog.getTableMetadata(sourceTable).copy(
-      identifier = targetTable,
-      tableType = CatalogTableType.MANAGED,
-      createTime = System.currentTimeMillis,
-      lastAccessTime = -1).withNewStorage(locationUri = None)
+    val sourceTableDesc = catalog.getTableMetadata(sourceTable)
+    val sourceTableType = sourceTableDesc.tableType
+
+    val newSerdeProp =
+      if (DDLUtils.isDatasourceTable(sourceTableDesc)) {
+        val newPath = catalog.defaultTablePath(targetTable)
+        sourceTableDesc.storage.properties.filterKeys(_.toLowerCase != "path") ++
+          Map("path" -> newPath)
+      } else {
+        sourceTableDesc.storage.properties
+      }
+
+    // Storage format
+    val newStorage =
+      if (sourceTableType == CatalogTableType.VIEW) {
```

Comment:
I see. No need to strictly follow Hive.

FYI, we are having two different default formats: https://github.com/apache/spark/pull/14430.